spyder/plugins/variableexplorer/widgets/dataframeeditor.py

```diff
@@ -1774,10 +1808,36 @@ def set_data_and_check(self, data) -> bool:
             self.table_header.setRowHeight(0, self.table_header.height())
 
         self.toolbar.clear()
-        for item in self.dataTable.menu_actions:
-            if item is not None:
-                if item.text() != 'Convert to':
-                    self.toolbar.addAction(item)
+        self.toolbar.addAction(self.dataTable.insert_action_above)
+        self.toolbar.addAction(self.dataTable.insert_action_below)
+        self.toolbar.addAction(self.dataTable.duplicate_row_action)
+        self.toolbar.addAction(self.dataTable.remove_row_action)
+        self.toolbar.addSeparator()
+        self.toolbar.addAction(self.dataTable.insert_action_before)
+        self.toolbar.addAction(self.dataTable.insert_action_after)
+        self.toolbar.addAction(self.dataTable.duplicate_col_action)
+        self.toolbar.addAction(self.dataTable.remove_col_action)
+
+        stretcher = self.create_stretcher('Toolbar stretcher')
+        self.toolbar.addWidget(stretcher)
+
+        self.toolbar.addAction(self.dataTable.resize_action)
+        self.toolbar.addAction(self.dataTable.resize_columns_action)
+        self.toolbar.addAction(self.refresh_action)
```

Comment:
And finally you'd need to add this actions (I'm not sure if you need to declare a section for them or not).